openwisp_utils/admin_theme/theme.py

```diff
@@ -0,0 +1,66 @@
+from django.core.exceptions import ImproperlyConfigured
+
+from . import settings as app_settings
+
+THEME_LINKS = app_settings.OPENWISP_ADMIN_THEME_LINKS.copy()
+THEME_JS = app_settings.OPENWISP_ADMIN_THEME_JS.copy()
+
+
+def register_theme_link(links):
+    if not isinstance(links, list):
+        raise ImproperlyConfigured(
+            '"openwisp_utils.admin_theme.theme.register_theme_link"'
+            ' accepts "list" of links.'
+        )
+    # We don't raise ImproperlyConfigured exception if
+    # link is already present in THEME_LINKS because
+    # user might specify a link in project settings which is
+    # already registered by an application.
+    # This would lead to time-consuming debugging for little returns.
```

Comment:
Are you sure this is consistent with what we do across all the openwisp codebase?

If I am not mistaken, in these functions which register new items in a data structure, we usually let it fail early and complaining loud if there's any duplicate registration attempted (which is a good thing and a best practice in the Python community to fail early and loud).

It doesn't look consistent with what we've been doing up to now to me.